claim-check-pattern/call-usage-app/src/main/java/com/iluwatar/claimcheckpattern/consumer/callcostprocessor/functions/UsageCostProcessorFunction.java

```diff
@@ -0,0 +1,167 @@
+        // Handle system events
+        if (eventGridEvent.getEventType()
+            .equals("Microsoft.EventGrid.SubscriptionValidationEvent")) {
+          SubscriptionValidationEventData subscriptionValidationEventData = eventGridEvent.getData()
+              .toObject(SubscriptionValidationEventData.class);
+          // Handle the subscription validation event
+          SubscriptionValidationResponse responseData = new SubscriptionValidationResponse();
+          responseData.setValidationResponse(subscriptionValidationEventData.getValidationCode());
+          return request.createResponseBuilder(HttpStatus.OK).body(responseData).build();
+
+        } else if (eventGridEvent.getEventType().equals("UsageDetail")) {
+          // Get message header and reference
+          MessageReference messageReference = eventGridEvent.getData()
+              .toObject(MessageReference.class);
+
+          // Read message from persistent storage
+          Message<UsageDetail> message = this.messageHandlerUtilityForUsageDetail
+              .readFromPersistantStorage(messageReference, context.getLogger());
+
+          // Get Data and generate cost details
+          List<UsageDetail> usageDetailsList = BinaryData.fromObject(
+              message.getMessageBody().getData())
+              .toObject(new TypeReference<>() {
+              });
+          List<UsageCostDetail> usageCostDetailsList = calculateUsageCostDetails(usageDetailsList);
+
+          // Create message body
+          MessageBody<UsageCostDetail> newMessageBody = new MessageBody<>();
+          newMessageBody.setData(usageCostDetailsList);
+
+          // Create message header
+          MessageReference newMessageReference = new MessageReference("callusageapp",
+              eventGridEvent.getId() + "/output.json");
+          MessageHeader newMessageHeader = new MessageHeader();
+          newMessageHeader.setId(eventGridEvent.getId());
+          newMessageHeader.setSubject("UsageCostProcessor");
+          newMessageHeader.setTopic("");
+          newMessageHeader.setEventType("UsageCostDetail");
+          newMessageHeader.setEventTime(OffsetDateTime.now().toString());
+          newMessageHeader.setData(newMessageReference);
+          newMessageHeader.setDataVersion("v1.0");
+
+          // Create entire message
+          Message<UsageCostDetail> newMessage = new Message<>();
+          newMessage.setMessageHeader(newMessageHeader);
+          newMessage.setMessageBody(newMessageBody);
+
+          // Drop data to persistent storage
+          this.messageHandlerUtilityForUsageCostDetail.dropToPersistantStorage(newMessage,
+              context.getLogger());
+
+          context.getLogger().info("Message is dropped successfully");
+          return request.createResponseBuilder(HttpStatus.OK)
+              .body("Message is dropped successfully").build();
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
Use a proper logger for output (see https://github.com/iluwatar/java-design-patterns/wiki/10.-Technology-selections)

Reply:
@iluwatar done

Reply:
@iluwatar removed